### PR TITLE
[JENKINS-39645] Upgrade to new parent POM

### PR DESCRIPTION
**Pull request by varyvol**
- Fix compilation errors in tests.
- Fix FindBugs errors.

[JENKINS-39645](https://issues.jenkins-ci.org/browse/JENKINS-39645)

@reviewbybees @olamy @imod

### CODE REVIEW

**Comment by imod** [comment]
most of this stuff is probably already in #18 which will be merged soon, please consider to add your changes on top of #18

File: src/main/java/org/jenkinsci/lib/configprovider/AbstractConfigProvider.java [hunk 1]
```diff
@@ -80,6 +81,7 @@ public void load() {
         }
     }
 
+    @SuppressFBWarnings(value = "NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE", justification = "Jenkins.getInstance() should never be null")
     protected XmlFile getConfigXml() {
         return new XmlFile(Jenkins.XSTREAM, new File(Jenkins.getInstance().getRootDir(), this.getXmlFileName()));
```
Comment (imod) [inline src/main/java/org/jenkinsci/lib/configprovider/AbstractConfigProvider.java]: a better way to fix this is by using `Jenkins.getActiveInstance()`

**Comment by varyvol** [comment]
@imod Done, I have merged the branch here. Basically only upgrading the version and fixing one test was necessary.

File: src/main/java/org/jenkinsci/plugins/configfiles/buildwrapper/ManagedFile.java [hunk 2]
```diff
@@ -79,11 +79,19 @@ public ManagedFile(String fileId, String targetLocation, String variable) {
         this.replaceTokens = false;
     }
 
+    public String getTargetLocation() {
```
Comment (imod) [inline src/main/java/org/jenkinsci/plugins/configfiles/buildwrapper/ManagedFile.java]: why do we need these getters? the fields are public already

**Pull request merged**